src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -237,12 +237,12 @@ class TaskBuilder(
 
         val containerWithPortMappings = portMappings match {
           case None => c
-          case Some(newMappings) => c.copy(
-            docker = c.docker.map {
-              _.copy(portMappings = newMappings)
-            }
-          )
+          case Some(newMappings) =>
+            c.foldDocker(c)(_.copy(
```

Comment:
Instead of mapping against the `container.docker(): Option[...]` above, it would be much more obvious to match against the container type here (then you don't need `foldDocker` btw), and provide a `def portMappings(docker: Container.Docker)` that computes the portMappings.